libtransact/src/contract/address/double_key_hash.rs

```diff
@@ -33,9 +33,9 @@ impl Addresser<(String, String)> for DoubleKeyHashAddresser {
         let hash_length = ADDRESS_LENGTH - self.prefix.len();
         let second_hash_length = hash_length - self.first_hash_length;
         if (self.prefix.len() + self.first_hash_length + second_hash_length) != ADDRESS_LENGTH {
-            return Err(AddresserError::DoubleKeyHashAddresserError(
-                "Invalid hash length".to_string(),
-            ));
+            return Err(AddresserError {
+                message: "Invalid hash length".to_string(),
```

Comment:
This message should be more informative.  It's also a bit of a late-binding error given that it would be based on the `first_hash_length` added in the constructor.   There should be some checks there.

Reply:
I added more validation for the values provided when constructing the Addressers, the changes are present in the last two commits I pushed up.